### PR TITLE
[JENKINS-70392] Kubernetes client 6.x follow-up compatibility

### PR DESCRIPTION
**Pull request by Vlatombe**
For compatibility with Kubernetes client 6.x

Also adds RJR based tests as `JenkinsRule` tests did not catch the classloading issue.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java [hunk 1]
```diff
@@ -184,4 +187,19 @@ private static StandardCredentials resolveCredentials(@CheckForNull String crede
         }
         return c;
     }
+
+    private static class WithContextClassLoader implements AutoCloseable {
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java:191]: This is available in core. (Not sure about your current baseline offhand, but at least leave a TODO.)

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/steps/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package org.csanchez.jenkins.plugins.kubernetes.pipeline.steps;
+/**
+ * Reusable steps for tests relying on RealJenkinsRule.
+ */
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/steps/package-info.java:4]: I think this is backwards? (Javadoc first, then `package` statement)

**Review by jglick — COMMENTED** [review]
Looks promising. What is missing in RJR to let this run in `kind.sh`?

**Comment by Vlatombe** [comment]
> What is missing in RJR to let this run in kind.sh?

See https://github.com/jenkinsci/kubernetes-plugin/pull/1290 for the follow-up (I needed to let this one go since the release was broken.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java (hunk 1 above)
Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java:191]: https://github.com/jenkinsci/kubernetes-plugin/pull/1289/files#diff-fe2f8caf9a928226f5d0a4ba20ff6a57846f75fcec2fcc925962cf87fd8f4040R147

**Comment by rpaasche** [comment]
I'm still getting this error with [3800.v54b_077b_94b_a_4](https://plugins.jenkins.io/kubernetes)
```
2023-01-10 15:49:01 |  hudson.remoting.ProxyException: io.fabric8.kubernetes.client.KubernetesClientException: No httpclient implementations found on the context classloader, please ensure your classpath includes an implementation jar
2023-01-10 15:49:01 |  	at io.fabric8.kubernetes.client.utils.HttpClientUtils.getHttpClientFactory(HttpClientUtils.java:164)
2023-01-10 15:49:01 |  	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:145)
2023-01-10 15:49:01 |  	at io.fabric8.kubernetes.client.DefaultKubernetesClient.<init>(DefaultKubernetesClient.java:56)
2023-01-10 15:49:01 |  	at io.fabric8.kubernetes.client.DefaultKubernetesClient.<init>(DefaultKubernetesClient.java:48)
2023-01-10 15:49:01 |  	at org.csanchez.jenkins.plugins.kubernetes.PodTemplateUtils.parseFromYaml(PodTemplateUtils.java:602)
2023-01-10 15:49:01 |  	at org.csanchez.jenkins.plugins.kubernetes.PodTemplateUtils.validateYamlContainerNames(PodTemplateUtils.java:639)
2023-01-10 15:49:01 |  	at org.csanchez.jenkins.plugins.kubernetes.PodTemplateUtils.validateYamlContainerNames(PodTemplateUtils.java:629)
2023-01-10 15:49:01 |  	at org.csanchez.jenkins.plugins.kubernetes.pipeline.PodTemplateStepExecution.start(PodTemplateStepExecution.java:145)
```

**Comment by mobidroid** [comment]
Getting the same error when trying to test connection from the Manage Nodes & Clouds > Kubernetes Cloud: 

`Error testing connection https://<my-cluster> No httpclient implementations found on the context classloader, please ensure your classpath includes an implementation jar`

**Comment by PatrickGhezali** [comment]
Same issue here with this plugin combination:

Kubernetes Client API Plugin _[6.3.1-206.v76d3b_6b_14db_b](https://plugins.jenkins.io/kubernetes-client-api)_
Kubernetes Credentials Plugin _[0.10.0](https://plugins.jenkins.io/kubernetes-credentials)_
Kubernetes plugin _[3800.v54b_077b_94b_a_4](https://plugins.jenkins.io/kubernetes)_

**Comment by Nuc7ear** [comment]
> Kubernetes Client API Plugin _[6.3.1-206.v76d3b_6b_14db_b](https://plugins.jenkins.io/kubernetes-client-api)_
> Kubernetes Credentials Plugin _[0.10.0](https://plugins.jenkins.io/kubernetes-credentials)_
> Kubernetes plugin _[3800.v54b_077b_94b_a_4](https://plugins.jenkins.io/kubernetes)_

Same problem with this version of plugins

**Comment by Vlatombe** [comment]
Very sorry for these inconvenience, seems like I missed a few more occurrences.

**Comment by PatrickGhezali** [comment]
Thanks for taking care!

**Comment by Vlatombe** [comment]
https://github.com/jenkinsci/kubernetes-plugin/pull/1291

**Pull request merged**